src/psyclone/psyir/symbols/datasymbol.py

```diff
@@ -171,13 +173,20 @@ def is_constant(self, value):
             constant.
 
         :raises ValueError: if `value` is True but this symbol does not have an
-            initial value set and does not have an ImportInterface.
+            initial value or an import or unresolved interface.
 
         '''
-        if value and not self.is_import and self.initial_value is None:
-            raise ValueError(
-                f"DataSymbol '{self.name}' does not have an initial value set "
-                f"and is not imported and therefore cannot be a constant.")
+        if (value and not (self.is_import or self.is_unresolved) and
+                self.initial_value is None):
+            # A Symbol of UnknownType could have initialisation within its
+            # original declaration.
+            # TODO could give TypedSymbol an 'is_unknown_type' property.
```

Comment:
@sergisiso: what would you prefer here - that I add an `is_unknown_type` property to Symbol or just add a pylint disable for the import?